libtrap/src/ifc_tcpip.c

```diff
@@ -1372,7 +1488,18 @@ int tcpip_sender_send(void *priv, const void *data, uint32_t size, int timeout)
 void tcpip_sender_terminate(void *priv)
 {
    tcpip_sender_private_t *c = (tcpip_sender_private_t *) priv;
+
+   uint32_t i;
+   uint64_t sum;
+
    if (c != NULL) {
+      do {
+         sum = 0;
+         for (i = 0; i < c->buffer_count; i++) {
+            sum |= c->buffers[i].clients_bit_arr;
+         }
+      } while (sum != 0);
```

Comment:
We are waiting for connected clients to receive all finished buffers (so that we dont lose messages when terminating TRAP). 

I put a 10 millisecond sleep there as suggested.